afqinsight/nn/pt_models.py

```diff
@@ -0,0 +1,401 @@
+        if self.output_activation:
+            x = self.output_activation(x)
+        return x
+
+def cnn_vgg(input_shape, n_classes):
+    cnn_vgg = cnn_vgg(input_shape, n_classes, output_activation = torch.softmax, verbose=False)
+    return cnn_vgg
+    
+class lstm1v0(nn.Module):
+    def __init__(self, input_shape, n_classes, output_activation = torch.softmax, verbose=False):
+        super(lstm1v0, self).__init__()
+        self.model = nn.Sequential(
+            nn.LSTM(input_shape[1], 512, batch_first=True),
+            nn.Linear(512, n_classes)
+        )
+        
+        if output_activation == torch.softmax:
+            self.output_activation = nn.Softmax(dim=1)
+        else:
+            self.output_activation = None
+
+    def forward(self, x):
+        x = self.model(x)
+        if self.output_activation:
+            x = self.output_activation(x)
+        return x
+
+def lstm1v0(input_shape, n_classes):
+    lstm1v0 = lstm1v0(input_shape, n_classes, output_activation = torch.softmax, verbose=False)
+    return lstm1v0
+    
+class lstm1(nn.Module):
+    def __init__(self, input_shape, n_classes, output_activation = torch.softmax, verbose=False):   
+        super(lstm1, self).__init__()
+        self.model = nn.Sequential(
+            nn.LSTM(input_shape[1], 100),
+            nn.ReLU(),
+            nn.Linear(100, n_classes)
+        )
+        
+        if output_activation == torch.softmax:
+            self.output_activation = nn.Softmax(dim=1)
+        else:
+            self.output_activation = None
+
+    def forward(self, x):
+        x = self.model(x)
+        if self.output_activation:
+            x = self.output_activation(x)
+        return x
+    
+def lstm1(input_shape, n_classes):
+    lstm1 = lstm1(input_shape, n_classes, output_activation = torch.softmax, verbose=False)
+    return lstm1
+    
+class lstm2(nn.Module):
+    def __init__(self, input_shape, n_classes, output_activation = torch.softmax, verbose=False):
+        super(lstm2, self).__init__()
+        self.model = nn.Sequential(
+            nn.LSTM(input_shape[1], 100),
+            nn.ReLU(),
+            nn.LSTM(100, 100),
+            nn.ReLU(),
+            nn.Linear(100, n_classes)
+        )
+        
+        if output_activation == torch.softmax:
+            self.output_activation = nn.Softmax(dim=1)
+        else:
+            self.output_activation = None
+
+    def forward(self, x):
+        x = self.model(x)
+        if self.output_activation:
+            x = self.output_activation(x)
+        return x
+
+def lstm2(input_shape, n_classes):
+    lstm2 = lstm2(input_shape, n_classes, output_activation = torch.softmax, verbose=False)
+    return lstm2
+    
+class blstm1(nn.Module):
+    def __init__(self, input_shape, n_classes, output_activation = torch.softmax, verbose=False):
+        super(blstm1, self).__init__()
+        self.model = nn.Sequential(
+            nn.LSTM(input_shape[1], 100, bidirectional=True),
```

Comment:
I think this is identical (or very similar) to blstm2? I think blstm1 is a lot simpler.